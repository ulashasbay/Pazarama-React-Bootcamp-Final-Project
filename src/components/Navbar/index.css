.navbar {
  background: linear-gradient(
    90deg,
    rgb(255, 255, 255) 0%,
    rgb(247, 246, 246) 100%
  );
  height: 80px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1.2rem;
}

.navbar-logo {
  justify-self: start;
  margin-left: 20px;
  cursor: pointer;
}

.fa-firstdraft {
  margin-left: 0.5rem;
  font-size: 1.6rem;
}

.nav-menu {
  display: grid;
  grid-template-columns: repeat(5, auto);
  grid-gap: 10px;
  list-style: none;
  text-align: center;
  width: 70vw;
  justify-content: end;
  margin-right: 2rem;
}

.nav-item {
  display: flex;
  align-items: center;
  height: 80px;
}

.nav-links {
  color: rgb(28, 28, 29);
  text-decoration: none;
  padding: 0.5rem 1rem;
  font-size: 16px;
}

.nav-links:hover {
  opacity: 0.8;
  transition: all 0.2s ease-out;
}

.fa-bars {
  color: #0137f3;
}

.nav-links-mobile {
  display: none;
}

.menu-icon {
  display: none;
}

@media screen and (max-width: 960px) {
  .nav-menu {
    display: flex;
    flex-direction: column;
    width: 100%;
    height: 90vh;
    position: absolute;
    top: 80px;
    left: -100%;
    opacity: 1;
    transition: all 0.5s ease;
  }

  .nav-menu.active {
    background: #faf9f9;
    left: 0;
    opacity: 1;
    transition: all 0.5s ease;
    z-index: 1;
  }

  .nav-links {
    text-align: center;
    padding: 2rem;
    width: 100%;
    display: table;
    font-size: 22px;
  }

  .nav-links:hover {
    background-color: #ff008b;
    color: rgb(255, 255, 255);
    border-radius: 0;
  }

  .navbar-logo {
    position: absolute;
    top: 8px;
    left: 0;
    transform: translate(25%, 50%);
  }

  .menu-icon {
    display: block;
    position: absolute;
    top: 4px;
    right: 0;
    transform: translate(-100%, 60%);
    font-size: 1.8rem;
    cursor: pointer;
  }

  .fa-times {
    color: #0137f3;
    font-size: 2rem;
  }

  .nav-links-mobile {
    display: block;
    text-align: center;
    padding: 1.5rem;
    margin: 2rem auto;
    border-radius: 4px;
    width: 80%;
    background: #0137f3;
    text-decoration: none;
    color: #fff;
    font-size: 1.5rem;
  }

  .nav-links-mobile:hover {
    background: #0037ff;
  }

  .navbar button {
    display: none;
  }
}
