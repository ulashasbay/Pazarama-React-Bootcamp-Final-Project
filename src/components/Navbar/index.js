import React, { useState } from "react";
import { Link } from "react-router-dom";
import { useNavigate } from "react-router-dom";
import NavbarBtn from "../NavbarBtn";
import PazaramaIcon from "../../assets/svg/PazaramaIcon";
import { useSelector } from "react-redux";
import { useDispatch } from "react-redux";
import { updateIsLoggedIn } from "../../redux/isLoggedIn/isLoggedInSlice";
import "./index.css";

function Navbar() {
  let navigate = useNavigate();

  const isLoggedInValue = useSelector((state) => state.isLoggedIn.value);
  const dispatch = useDispatch();

  const [click, setClick] = useState(false);

  const handleClick = () => setClick(!click);

  const closeMobileMenu = () => setClick(false);

  const handleLogout = () => {
    localStorage.removeItem("username");
    localStorage.removeItem("password");
    dispatch(updateIsLoggedIn(false));
    navigate("/");
    closeMobileMenu();
  };

  return (
    <>
      <nav className="navbar">
        <Link to="/" className="navbar-logo" onClick={closeMobileMenu}>
          <PazaramaIcon />
        </Link>
        <div className="menu-icon" onClick={handleClick}>
          <i className={click ? "fas fa-times" : "fas fa-bars"} />
        </div>
        <ul className={click ? "nav-menu active" : "nav-menu"}>
          <li className="nav-item">
            <Link to="/" className="nav-links" onClick={closeMobileMenu}>
              Anasayfa
            </Link>
          </li>
          <li className="nav-item">
            <Link
              to="/basvuru-olustur"
              className="nav-links"
              onClick={closeMobileMenu}
            >
              Başvuru Yap
            </Link>
          </li>
          {isLoggedInValue && (
            <li className="nav-item">
              <Link
                to="/admin/basvuru-listesi"
                className="nav-links"
                onClick={closeMobileMenu}
              >
                Başvuru Listesi
              </Link>
            </li>
          )}
          <li className="nav-item">
            <Link
              to="/basvuru-sorgula"
              className="nav-links"
              onClick={closeMobileMenu}
            >
              Başvuru Sorgula
            </Link>
          </li>
          {!isLoggedInValue ? (
            <li>
              <Link
                to="/admin"
                className="nav-links-mobile"
                onClick={closeMobileMenu}
              >
                Giriş
              </Link>
            </li>
          ) : (
            <li>
              <Link to="/" className="nav-links-mobile" onClick={handleLogout}>
                Çıkış
              </Link>
            </li>
          )}
        </ul>

        {!isLoggedInValue ? (
          <NavbarBtn text="Giriş" onClick={() => navigate("/admin")} />
        ) : (
          <NavbarBtn text="Çıkış" onClick={handleLogout} />
        )}
      </nav>
    </>
  );
}

export default Navbar;
