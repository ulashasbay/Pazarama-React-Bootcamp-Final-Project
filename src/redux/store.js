import { configureStore } from "@reduxjs/toolkit";
import appIdReducer from "./appId/appIdSlice";
import userInfoReducer from "./userInfo/userInfoSlice";
import isLoggedInReducer from "./isLoggedIn/isLoggedInSlice";

export const store = configureStore({
  reducer: {
    appId: appIdReducer,
    userInfo: userInfoReducer,
    isLoggedIn: isLoggedInReducer,
  },
});
