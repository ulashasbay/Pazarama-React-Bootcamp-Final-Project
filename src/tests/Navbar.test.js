import React from "react";
import { render, screen } from "@testing-library/react";
import Navbar from "../components/Navbar";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../redux/store";
import { createMemoryHistory } from "history";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";

const AllTheProviders = ({ children }) => {
  return (
    <BrowserRouter>
      <Provider store={store}>{children}</Provider>
    </BrowserRouter>
  );
};

describe("Navbar Test", () => {
  let anasayfaLink, basvuruYapLink, basvuruListesiLink, basvuruSorgulaLink;
  beforeEach(() => {
    render(<Navbar />, { wrapper: AllTheProviders });
    anasayfaLink = screen.getByText("Anasayfa");
    basvuruYapLink = screen.getByText("Başvuru Yap");
    basvuruListesiLink = screen.getByText("Başvuru Listesi");
    basvuruSorgulaLink = screen.getByText("Başvuru Sorgula");
  });

  test("Render Navbar Menu items", () => {
    expect(anasayfaLink).toBeInTheDocument();
    expect(basvuruYapLink).toBeInTheDocument();
    expect(basvuruListesiLink).toBeInTheDocument();
    expect(basvuruSorgulaLink).toBeInTheDocument();
  });

  test("Render Navbar Buttons", () => {
    const btns = screen.getByRole("button");
    expect(btns).toBeInTheDocument();
  });

  test("Clicking Anasayfa should redirect", () => {
    const history = createMemoryHistory();
    userEvent.click(anasayfaLink);
    expect(history.location.pathname).toEqual("/");
  });

  test("Clicking Başvuru Yap should redirect to /basvuru-olustur", () => {
    userEvent.click(basvuruYapLink);
    expect(location.pathname).toEqual("/basvuru-olustur");
  });

  test("Clicking Başvuru Listesi should redirect to /admin/basvuru-listesi", () => {
    userEvent.click(basvuruListesiLink);
    expect(location.pathname).toEqual("/admin/basvuru-listesi");
  });

  test("Clicking Başvuru Sorgula should redirect to /basvuru-sorgula", () => {
    userEvent.click(basvuruSorgulaLink);
    expect(location.pathname).toEqual("/basvuru-sorgula");
  });
});

test("Navbar renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <Navbar />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
