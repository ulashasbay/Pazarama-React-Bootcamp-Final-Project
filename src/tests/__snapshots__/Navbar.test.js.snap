// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  className="navbar"
>
  <a
    className="navbar-logo"
    href="/"
    onClick={[Function]}
  >
    <svg
      height="2rem"
      viewBox="0 0 170.081 27.276"
      width="9rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.679 1.314a9.764 9.764 0 0 0-3.6 3.6 9.8 9.8 0 0 0-1.333 4.994 9.73 9.73 0 0 0 1.333 4.975 9.764 9.764 0 0 0 3.6 3.6 9.8 9.8 0 0 0 4.994 1.333 9.8 9.8 0 0 0 4.994-1.333 9.764 9.764 0 0 0 3.6-3.6 9.714 9.714 0 0 0 1.314-4.975 9.782 9.782 0 0 0-1.314-4.994 9.764 9.764 0 0 0-3.6-3.6A9.782 9.782 0 0 0 13.673 0a9.782 9.782 0 0 0-4.994 1.314zm9.405 4.13a6.285 6.285 0 0 1 1.727 4.468 6.208 6.208 0 0 1-1.746 4.449 5.754 5.754 0 0 1-4.355 1.821 5.864 5.864 0 0 1-4.392-1.821 6.1 6.1 0 0 1-1.8-4.449 6.164 6.164 0 0 1 1.783-4.449 5.906 5.906 0 0 1 4.412-1.821 5.847 5.847 0 0 1 4.371 1.802z"
        data-name="Path 73"
        fill="#f219af"
      />
      <path
        d="M-4.571 2.9h3.818V-7.277a7.609 7.609 0 0 0 6.095 2.765 9.8 9.8 0 0 0 4.985-1.335 9.784 9.784 0 0 0 3.611-3.611 9.749 9.749 0 0 0 1.335-4.985 9.817 9.817 0 0 0-1.335-5 9.784 9.784 0 0 0-3.611-3.611 9.782 9.782 0 0 0-4.985-1.317 9.918 9.918 0 0 0-9.913 9.932zM5.342-8.161A5.864 5.864 0 0 1 .959-9.986a6.176 6.176 0 0 1-1.787-4.458A6.176 6.176 0 0 1 .959-18.9a5.835 5.835 0 0 1 4.383-1.825 5.93 5.93 0 0 1 4.4 1.806 6.234 6.234 0 0 1 1.749 4.477 6.159 6.159 0 0 1-1.767 4.456 5.835 5.835 0 0 1-4.382 1.825zm26.616 3.649a7.4 7.4 0 0 0 3.706-.959 5.565 5.565 0 0 0 2.408-2.389v3.047h3.781v-9.631a9.8 9.8 0 0 0-1.317-5 9.784 9.784 0 0 0-3.611-3.611 9.714 9.714 0 0 0-4.966-1.317 9.782 9.782 0 0 0-4.985 1.317 9.834 9.834 0 0 0-3.63 3.611 9.817 9.817 0 0 0-1.335 5 9.95 9.95 0 0 0 9.949 9.932zm-4.364-5.474a6.117 6.117 0 0 1-1.806-4.458 6.176 6.176 0 0 1 1.787-4.456A5.917 5.917 0 0 1 32-20.726a5.859 5.859 0 0 1 4.383 1.806 6.3 6.3 0 0 1 1.731 4.477 6.221 6.221 0 0 1-1.749 4.458A5.765 5.765 0 0 1 32-8.161a5.876 5.876 0 0 1-4.406-1.825zm31.28-10.421v-3.668H44.316v3.668h9.743S44.558-9.458 44.013-8.549A2.452 2.452 0 0 0 43.77-7.3c-.009 1.285 0 2.484 0 2.484h15.443v-3.665H48.548zM69.746-4.512a7.4 7.4 0 0 0 3.706-.959 5.565 5.565 0 0 0 2.408-2.39v3.047h3.78v-9.631a9.8 9.8 0 0 0-1.317-5 9.784 9.784 0 0 0-3.611-3.611 9.714 9.714 0 0 0-4.966-1.317 9.782 9.782 0 0 0-4.985 1.317 9.834 9.834 0 0 0-3.63 3.611 9.817 9.817 0 0 0-1.335 5 9.95 9.95 0 0 0 9.95 9.933zm-4.363-5.474a6.117 6.117 0 0 1-1.806-4.458 6.176 6.176 0 0 1 1.787-4.456 5.917 5.917 0 0 1 4.42-1.825 5.859 5.859 0 0 1 4.383 1.806 6.3 6.3 0 0 1 1.733 4.475 6.221 6.221 0 0 1-1.749 4.458 5.765 5.765 0 0 1-4.364 1.825 5.876 5.876 0 0 1-4.404-1.825zm20.71 5.173v-10.064c0-3.781 1.561-5.624 4.326-5.624h.376v-3.649h-.376a7.736 7.736 0 0 0-5.887 2.332c-1.486 1.561-2.22 3.875-2.22 6.941v10.064zm15.612.3a7.4 7.4 0 0 0 3.706-.959 5.565 5.565 0 0 0 2.408-2.389v3.047h3.781v-9.631a9.8 9.8 0 0 0-1.317-5 9.784 9.784 0 0 0-3.611-3.611 9.714 9.714 0 0 0-4.966-1.317 9.782 9.782 0 0 0-4.985 1.317 9.834 9.834 0 0 0-3.63 3.611 9.817 9.817 0 0 0-1.335 5 9.95 9.95 0 0 0 9.944 9.933zM97.34-9.986a6.117 6.117 0 0 1-1.806-4.458 6.176 6.176 0 0 1 1.788-4.456 5.917 5.917 0 0 1 4.42-1.825 5.859 5.859 0 0 1 4.383 1.806 6.3 6.3 0 0 1 1.731 4.477 6.221 6.221 0 0 1-1.749 4.458 5.765 5.765 0 0 1-4.364 1.825 5.876 5.876 0 0 1-4.403-1.827zm37.86-14.389a6.96 6.96 0 0 0-6.471 3.574 6.96 6.96 0 0 0-6.471-3.574 8.061 8.061 0 0 0-6.076 2.37c-1.524 1.6-2.295 3.95-2.295 7.054v10.138h3.781v-10.139c0-3.856 1.636-5.775 4.59-5.775 3.066 0 4.59 1.919 4.59 5.775v10.139h3.762v-10.139c0-3.856 1.636-5.775 4.59-5.775 3.066 0 4.59 1.919 4.59 5.775v10.139h3.762v-10.139c0-3.1-.752-5.455-2.276-7.054a8.061 8.061 0 0 0-6.076-2.369zm20.413 19.863a7.4 7.4 0 0 0 3.706-.959 5.565 5.565 0 0 0 2.408-2.389v3.047h3.781v-9.631a9.8 9.8 0 0 0-1.317-5 9.784 9.784 0 0 0-3.611-3.611 9.714 9.714 0 0 0-4.966-1.317 9.782 9.782 0 0 0-4.985 1.317A9.834 9.834 0 0 0 147-19.447a9.818 9.818 0 0 0-1.335 5 9.95 9.95 0 0 0 9.948 9.935zm-4.364-5.474a6.117 6.117 0 0 1-1.806-4.458 6.176 6.176 0 0 1 1.787-4.456 5.917 5.917 0 0 1 4.42-1.825 5.859 5.859 0 0 1 4.383 1.806 6.3 6.3 0 0 1 1.73 4.477 6.221 6.221 0 0 1-1.749 4.458 5.765 5.765 0 0 1-4.364 1.825 5.876 5.876 0 0 1-4.401-1.827z"
        data-name="Path 70"
        style={
          Object {
            "fill": "#0136f4",
          }
        }
        transform="translate(4.574 24.378)"
      />
      <path
        d="M7.518 9.744H3.744A9.852 9.852 0 0 1 13.674 0a9.827 9.827 0 0 1 9.91 9.742h-3.676a6.378 6.378 0 0 0-1.823-4.3 5.865 5.865 0 0 0-4.374-1.8 5.918 5.918 0 0 0-4.412 1.821 6.168 6.168 0 0 0-1.781 4.28z"
        data-name="Subtraction 1"
        fill="#f219af"
      />
    </svg>
  </a>
  <div
    className="menu-icon"
    onClick={[Function]}
  >
    <i
      className="fas fa-bars"
    />
  </div>
  <ul
    className="nav-menu"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-links"
        href="/"
        onClick={[Function]}
      >
        Anasayfa
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-links"
        href="/basvuru-olustur"
        onClick={[Function]}
      >
        Başvuru Yap
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-links"
        href="/admin/basvuru-listesi"
        onClick={[Function]}
      >
        Başvuru Listesi
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-links"
        href="/basvuru-sorgula"
        onClick={[Function]}
      >
        Başvuru Sorgula
      </a>
    </li>
    <li>
      <a
        className="nav-links-mobile"
        href="/"
        onClick={[Function]}
      >
        Çıkış
      </a>
    </li>
  </ul>
  <button
    className="btn"
    onClick={[Function]}
  >
    Çıkış
  </button>
</nav>
`;
